### PR TITLE
chore(bytes): release `bytes@1.0.0`

### LINKED ISSUE #4629: proposal: stabilize `std/bytes`
This is the first proposal to stabilize a package within the Standard Library, so let's aim to improve the process. This process depends on first fully transferring the Standard Library to [JSR](https://jsr.io/). This issue is meant for tracking and discussing making the package stable.

Requirements:

- [x] 100% documented, passing deno doc --lint checks and adhering to the [documentation guidelines](https://github.com/denoland/deno_std/blob/main/.github/CONTRIBUTING.md#documentation).
- [x] Maximum possible test coverage
- [x] No open issues or pull requests that might lead to breaking changes. For example, issues that suggest new non-breaking features are fine to exist at stabilization.
- [x] Release `1.0.0-rc.1`, approved by 2 members of the internal team.
- [ ] Feedback month (May 7 - June 7)
- [ ] (Scheduled June 7) Stabilization

Blockers:

- [x] #3603

### PR DESCRIPTION
**Pull request by iuioiua**
To be merged on June ~~7~~ 6.

Feedback from all is welcome before this package is stabilized.

Closes #4629

### CODE REVIEW

**Comment by kt3k** [comment]
Do we need this PR at this point? How about getting feedback in #4629?

**Comment by iuioiua** [comment]
A PR that stabilizes the package is far more obvious and likely to receive feedback. It's a little ugly to have PRs open for so long, but doing so hopefully increases our chances of an issue being pointed out.

File: bytes/deno.json
Comment (dsherret) [inline bytes/deno.json:1]: It looks like `indexOfNeedle` with a negative start value is not tested.

**Review by dsherret — APPROVED** [review]
LGTM

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**